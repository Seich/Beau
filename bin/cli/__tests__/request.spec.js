const RequestCommand = require('../commands/request');
const requestPromiseNativeMock = require('request-promise-native');

jest.mock('../../../src/shared');

jest.mock('../base');

describe('Request Command', () => {
    let result;

    beforeEach(() => {
        requestPromiseNativeMock.fail = false;
        result = [];
        jest.spyOn(process.stdout, 'write').mockImplementation(val =>
            result.push(require('strip-ansi')(val.toString('utf8')))
        );
    });

    afterEach(() => jest.restoreAllMocks());

    test.each([
        ['alias'],
        ['alias', '--verbose'],
        ['alias', '--as-json'],
        ['alias', '--as-json', '--verbose'],
        ['alias', '--no-format'],
        ['alias', '--quiet']
    ])('with flags: %s %s %s', async (...args) => {
        await RequestCommand.run(args);
        expect(result).toMatchSnapshot();
    });

    it('should throw an error when the request fails', async () => {
        requestPromiseNativeMock.fail = true;
        await expect(RequestCommand.run(['anything'])).rejects.toThrow(Error);
    });
});
