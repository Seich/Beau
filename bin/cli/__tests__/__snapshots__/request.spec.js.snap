// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Command with flags: alias %s %s 1`] = `
Array [
  "",
  "  Status              Endpoint            
",
  "  200                 https://example.org/anything
",
  "
",
  "\\"{\\"hello\\": \\"world\\"}\\"
",
]
`;

exports[`Request Command with flags: alias --as-json %s 1`] = `
Array [
  "{\\"status\\":200,\\"headers\\":[],\\"body\\":\\"{\\\\\\"hello\\\\\\": \\\\\\"world\\\\\\"}\\"}
",
]
`;

exports[`Request Command with flags: alias --as-json --verbose 1`] = `
Array [
  "{\\"request\\":{\\"body\\":{\\"name\\":\\"David\\"},\\"endpoint\\":\\"https://example.org/anything\\"},\\"response\\":{\\"status\\":200,\\"headers\\":[],\\"body\\":\\"{\\\\\\"hello\\\\\\": \\\\\\"world\\\\\\"}\\"},\\"body\\":\\"{\\\\\\"hello\\\\\\": \\\\\\"world\\\\\\"}\\"}
",
]
`;

exports[`Request Command with flags: alias --no-format %s 1`] = `
Array [
  "200
",
  "https://example.org/anything
",
  "[]
",
  "\\"{\\\\\\"hello\\\\\\": \\\\\\"world\\\\\\"}\\"
",
]
`;

exports[`Request Command with flags: alias --quiet %s 1`] = `Array []`;

exports[`Request Command with flags: alias --verbose %s 1`] = `
Array [
  "",
  "  Status              Endpoint            
",
  "  200                 https://example.org/anything
",
  "
",
  "{
  request: {
    body: {
      name: \\"David\\"
    },
    endpoint: \\"https://example.org/anything\\"
  },
  response: {
    status: 200,
    headers: [],
    body: \\"{\\"hello\\": \\"world\\"}\\"
  },
  body: \\"{\\"hello\\": \\"world\\"}\\"
}
",
]
`;
